gnucash/report/standard-reports/transaction.scm

```diff
@@ -1157,16 +1157,14 @@ be excluded from periodic reporting.")
                        (vector (_ "Price")
                                (lambda (split transaction-row?)
                                  ;; share price is retrieved as an exact rational; convert for
-                                 ;; presentation to decimal, rounded to the currency SCU, optionally
-                                 ;; increasing precision by 2 significant digits.
+                                 ;; presentation to decimal, rounded to the currency SCU (inc by 2
+                                 ;; decimal places), with a minimum of 4 decimal places.
                                  (let* ((currency (xaccTransGetCurrency (xaccSplitGetParent split)))
                                         (scu (gnc-commodity-get-fraction currency))
                                         (price (xaccSplitGetSharePrice split))
-                                        (price-decimal (gnc-numeric-convert price
-                                                                            (if (< scu 10000)
-                                                                                (* scu 100)
-                                                                                scu)
-                                                                            GNC-HOW-RND-ROUND)))
+                                        (price-decimal (gnc-numeric-convert
+                                                        price (max (* 100 scu) 10000)
+                                                        GNC-HOW-RND-ROUND)))
```

Comment:
The commit description doesn't describe what the code does, which is set the price's displayed precision to at least 4 decimal places and at most the smallest currency unit's decimal places plus two more. Bitcoin and possibly other e-currencies has an SCU of 10^-9, so prices in BTC will be displayed with 11 decimal places. This does that. Japanese Yen, on the other hand, have an SCU of 1: It's all pennies. I'm not sure that anything is gained by displaying prices in JPY to 4 rather than 2 decimal places, but I'm also not sure that much is gained by displaying BTC prices with 11 instead of 9.